.github/workflows/security.yml

```
@@ -0,0 +1,167 @@
+      with:
+        fetch-depth: 0  # Full history for scanning
+
+    - name: Install Gitleaks
+      run: |
+        wget -q https://github.com/gitleaks/gitleaks/releases/download/v8.18.1/gitleaks_8.18.1_linux_x64.tar.gz
+        tar -xzf gitleaks_8.18.1_linux_x64.tar.gz
+        chmod +x gitleaks
+
+    - name: Run Gitleaks
+      id: gitleaks
+      run: |
```

actionlint (1.7.10)

shellcheck reported issue in this script: SC2086:info:2:120: Double quote to prevent globbing and word splitting

(shellcheck)